### DISCUSSION TITLE
adding proteins to existing clusters

### DISCUSSION THREAD

**Discussion by nohayoussef** [micans/mcl#24] [Q&A] [answered]
Hi,
I have created clusters from blast output with a subset of proteins (all versus all blastp from ~800 genomes) that I am working with. Now I need to add more proteins (I expanded to ~2000 genomes including the original 800) to the existing clusters (plus of course add more clusters if needed) using new blast outputs. I have created 2 blast output: the ~1200 extra genomes proteins versus all the ~2000 genome proteins, as well as the ~800 original against the extra ~1200. The three blast outputs can of course be combined in one huge file and be formatted to the abc format. But my question is: can I keep adding to the original clustering file, or do I just need to start all over? Thanks for your time

**Comment by micans** [comment, marked as answer]
Hi, normal procedure is indeed to start all over and re-cluster everything. This is really also what you want most of the time, taking into account how everything relates to everything rather than depend on/extend from a foothold on a subset of the data.
Computationally this can be quite painful of course. In your case, the step from 800 to 2000 is really a big one; it would be a bit different if it were from 2000 to 2001.
Second, within the mcl software there is no support really if you want to perform such an extension step. It would be quite fiddly and involve a lot of choices. I do believe it is something people have done in the past (mapping new genomes onto existing clusters), but this is something that is really not particular to mcl and just involves projection/mapping. Suppose you went from 2000 to 2002 genomes, then this would be done by projecting the two extra genomes onto the existing clusters, and would probably ignore the within-similarities between the two new genomes (so without any aspect of clustering the new set of genomes).
